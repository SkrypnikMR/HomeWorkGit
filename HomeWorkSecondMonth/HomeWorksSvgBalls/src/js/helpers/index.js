import * as bumps from './bumps';
import * as render from './render';

export default { bumps, render};