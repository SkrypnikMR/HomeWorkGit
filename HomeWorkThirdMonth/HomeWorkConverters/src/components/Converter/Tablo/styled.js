import styled from 'styled-components';

export const StTablo = styled.div`
    ${({ transition = 'all 5000ms ease-in-out' }) => transition && `transition: ${transition}`};
    height: 50px;
    margin-top: 30px;
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    font-family: fantasy;
    font-size: 30px;
`;
export const StP = styled.p`
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
`;
export const StSpan = styled.span`
    ${({ color = 'blue' }) => color && `color: ${color}`};
    padding: 5px;
`;
