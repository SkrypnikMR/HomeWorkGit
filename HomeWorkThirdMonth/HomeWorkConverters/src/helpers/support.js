export const support = {
    lsGet: name => localStorage.getItem(name),
    lsSet: (name, item) => {
        if (typeof item === 'string') return localStorage.setItem(name, item);
        localStorage.setItem(name, JSON.stringify(item));
    },
    convert: (num, primary, secondary) => {
        const lengthSystem = {
            meters: {
                meters: 1,
                miles: 1609.344,
                yards: 1.093613,
                feet: 3.28084,
                versts: 1066.8,
            },
            miles: {
                meters: 1609.344,
                miles: 1,
                yards: 1760,
                feet: 0.003281,
                versts: 1.50857,
            },
            yards: {
                meters: 0.9144,
                miles: 0.000568182,
                yards: 1,
                feet: 3,
                versts: 0.000857143,
            },
            feet: {
                meters: 0.3048,
                miles: 0.000189394,
                yards: 0.333333,
                feet: 1,
                versts: 0.000285714,
            },
            versts: {
                meters: 1066.8,
                miles: 0.662879,
                yards: 1166.67,
                feet: 3500,
                versts: 1,
            },
        };
        return (+num * lengthSystem[primary][secondary]).toString();
    },
};
