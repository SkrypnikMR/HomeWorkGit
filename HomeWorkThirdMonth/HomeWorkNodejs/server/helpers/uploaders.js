require('dotenv/config');
const uuid = require('uuid/v4');
const AWS = require('aws-sdk');
const sharp = require('sharp');
const multer = require('multer');
const s3 = new AWS.S3({
    accessKeyId: process.env.AWS_ID,
    secretAccessKey: process.env.AWS_SECRET,
})

const fileUploader = async (req, res) => {
    const myFile = req.file.originalname.split('.');
    const fileType = myFile[myFile.length - 1];
    try {
        const params = {
            Bucket: process.env.AWS_BUCKET_NAME_FILES,
            Key: `file-${myFile[0]}-${uuid()}.${fileType}`,
            Body: req.file.buffer
        }
        const result = await s3.upload(params).promise().catch((e) => res.status(500).json({ message: 'server error' }));
        return res.status(200).json({ fileLink: result.Location });
    } catch (e) { res.status(500).json({ message: 'server error' }) }
}

const imageUploader = async (req, size, name) => {
    try {
        const myFile = req.file.originalname.split('.');
        const fileType = myFile[myFile.length - 1];

        const sizedImage = await sharp(req.file.buffer).resize({ width: size, height: size }).toBuffer();
        const params = { Bucket: process.env.AWS_BUCKET_NAME_IMAGES, Key: `${name}${myFile[0]}-${uuid()}.${fileType}`, Body: sizedImage }
        const result = await s3.upload(params).promise();

        return result.Location;
    } catch (e) {
        res.status(500).json({ message: 'server error' })
    }

}
const storage = multer.memoryStorage({ destination: (req, res, cb) => cb(null, '') });
const upload = multer({ storage, limits: { fileSize: 20971520 } }).single('file');

const corsOptions = {
    methods: ['POST', 'OPTIONS', 'GET'],
    origins: '*',
    allowedHeaders: ['Content-Type', 'multipart/form-data']
}

module.exports = { fileUploader, imageUploader, upload}