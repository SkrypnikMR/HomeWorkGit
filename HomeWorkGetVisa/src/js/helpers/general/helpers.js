import * as validate from "../validators/validators.js";
import { notifKill } from "../listeners/listeners.js";
import * as drowers from "../drowers/drowers.js";

export const click = (buttons) => {
  for (let key in buttons) {
    buttons[key].click();
  }
};

export const validator = (candidate) => {
  const vName = validate.validateName(candidate.name);
  const vBalance = validate.validateNumber(candidate.balance);
  const vAgelikeNumber = validate.validateNumber(candidate.age);
  const VAgeLength = validate.validateLength(candidate.age, 3);
  const vEnglish = validate.validateEnglish(candidate.english);
  if (vName || !vBalance || !vAgelikeNumber || VAgeLength || !vEnglish) {
    return false;
  }
  return true;
};

export const cleaner = (nodes) => {
  for (let key in nodes) {
    if (nodes[key].checked === true) {
      nodes[key].checked = false;
      continue;
    }
    nodes[key].value = "";
  }
};

export const createNotif = (str) => {
  // create notif Div
  const notif = document.createElement("div");
  const notifTextDiv = document.createElement("div");
  const notifButtonDiv = document.createElement("div");
  const notifDeleteBtn = document.createElement("button");
  const notifMessage = document.createElement("p");

  // add classes
  notif.classList.add("notification");
  notifButtonDiv.classList.add("notification__button");
  notifTextDiv.classList.add("notification__message");
  notifDeleteBtn.classList.add("button__item");
  notifDeleteBtn.classList.add("button");
  notifMessage.classList.add("message__item");

  //add textContent
  notifDeleteBtn.textContent = "X";
  notifMessage.textContent = str;

  //append chields
  notifTextDiv.appendChild(notifMessage);
  notifButtonDiv.appendChild(notifDeleteBtn);
  notif.appendChild(notifButtonDiv);
  notif.appendChild(notifTextDiv);
  document.body.appendChild(notif);

  // add event Listener
  notifDeleteBtn.addEventListener("click", notifKill.bind(null, notif));

  // autoDeath after 3s
  setTimeout(notifKill.bind(null, notif), 2000);
};

export const mutateObject = (obj) => {
  obj.balance = Number(obj.balance);
  obj.age = Number(obj.age);
  return obj;
};

export const drowInit = (candidates, canvas, context, distance_Between) => {
  drowers.drowBackground(canvas, context);
  for (let i = 0; i < candidates.length; i++) {
    const distance = distance_Between * (i + 1) + 2;
    const x = 0;
    const k = (2 * Math.PI) / 100;
    drowers.drowText(context, candidates[i].name, distance, 92, "white", 12);
    drowers.drowText(context, "balance", distance, 292, "white", 12);
    drowers.drowText(context, "documents", distance, 492, "white", 12);
    drowers.drowText(context, "age", distance, 692, "white", 12);
    drowers.drowText(context, "english", distance, 892, "white", 12);
    drowers.drowCircle(context, canvas, x, k, 100, distance, 100, 0, "green");
    candidates[i].distance = distance;
    candidates[i].balancePoint = 292;
    candidates[i].documentsPoint = 492;
    candidates[i].agePoint = 692;
    candidates[i].englishPoint = 892;
  }
};

export const sessionStorageSet = (key, value) =>
  sessionStorage.setItem(key, value);
