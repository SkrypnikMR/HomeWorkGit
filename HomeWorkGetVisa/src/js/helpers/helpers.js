import * as validate from "./validators/validators.js";
import { notifKill } from "./listeners.js";

export const click = (buttons) => {
  for (let key in buttons) {
    buttons[key].click();
  }
};

export const validator = (candidate) => {
  const vName = validate.validateName(candidate.name);
  const vBalance = validate.validateNumber(candidate.balance);
  const vAgelikeNumber = validate.validateNumber(candidate.age);
  const VAgeLength = validate.validateLength(candidate.age, 3);
  const vEnglish = validate.validateEnglish(candidate.english);
  if (vName || !vBalance || !vAgelikeNumber || VAgeLength || !vEnglish) {
    return false;
  }
  return true;
};

export const cleaner = (nodes) => {
  for (let key in nodes) {
    if (nodes[key].checked === true) {
      nodes[key].checked = false;
      continue;
    }
    nodes[key].value = "";
  }
};

export const createNotif = (str) => {
  // create notif Div
  const notif = document.createElement("div");
  const notifTextDiv = document.createElement("div");
  const notifButtonDiv = document.createElement("div");
  const notifDeleteBtn = document.createElement("button");
  const notifMessage = document.createElement("p");

  // add classes
  notif.classList.add("notification");
  notifButtonDiv.classList.add("notification__button");
  notifTextDiv.classList.add("notification__message");
  notifDeleteBtn.classList.add("button__item");
  notifDeleteBtn.classList.add("button");
  notifMessage.classList.add("message__item");

  //add textContent
  notifDeleteBtn.textContent = "X";
  notifMessage.textContent = str;

  //append chields
  notifTextDiv.appendChild(notifMessage);
  notifButtonDiv.appendChild(notifDeleteBtn);
  notif.appendChild(notifButtonDiv);
  notif.appendChild(notifTextDiv);
  document.body.appendChild(notif);

  // add event Listener
  notifDeleteBtn.addEventListener("click", notifKill.bind(null, notif));
};

export const mutateObject = (obj) => {
  obj.balance = Number(obj.balance);
  obj.age = Number(obj.age);
  return obj;
};

export const drow = (candidates, canvas, context) => {
  var box_Height = canvas.height;
  context.fillStyle = "rgba(94, 144, 252, 1)";
  context.fillRect(1, 1, canvas.width - 1, box_Height - 1);
  context.strokeRect(0.5, 0.5, canvas.width - 1, box_Height);
  const centerX = 100;
  const radius = 40;
  const distance_Between = 105;
  for (let i = 0; i < candidates.length; i++) {
    context.beginPath();
    context.font = " 12px Arial";
    context.fillStyle = "white";
    context.fillText(
      candidates[i].name,
      92 - candidates[i].name.length * 2,
      distance_Between * (i + 1) + 2
    );
    context.closePath();
    context.beginPath();
    context.font = " 12px Arial";
    context.fillStyle = "white";
    context.fillText(
      `check balance`,
      300 * 1.5,
      distance_Between * (i + 1) + 2
    );
    context.closePath();
    context.beginPath();
    context.font = " 12px Arial";
    context.fillStyle = "white";
    context.fillText(`check age`, 500 * 1.5, distance_Between * (i + 1) + 2);
    context.closePath();
    context.beginPath();
    context.font = " 12px Arial";
    context.fillStyle = "white";
    context.fillText(
      `check documents`,
      700 * 1.5,
      distance_Between * (i + 1) + 2
    );
    context.closePath();
    context.beginPath();
    context.font = " 12px Arial";
    context.fillStyle = "white";
    context.fillText(
      `check english`,
      900 * 1.5,
      distance_Between * (i + 1) + 2
    );
    context.closePath();
    let x = 0;
    let k = (2 * Math.PI) / 100;
    const kek = distance_Between * (i + 1) + 2;
    console.log(kek);
    drowCircle(context, canvas, x, k, kek);
  }
};

export function drowCircle(context, canvas, x, k, kek) {
  context.beginPath();
  context.arc(100, kek, 50, 0, x * k, false);
  context.lineWidth = 3;
  context.lineCap = "butt";
  context.strokeStyle = "green";
  context.stroke();
  context.closePath();
  let timeout;
  if (x <= 100) {
    timeout = setTimeout(() => {
      drowCircle(context, canvas, x + 1, k, kek);
    }, 50);
  } else clearTimeout(timeout);
}
