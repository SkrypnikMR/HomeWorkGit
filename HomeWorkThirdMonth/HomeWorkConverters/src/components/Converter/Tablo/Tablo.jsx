import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { StTablo, StP, StSpan } from './styled';
import { support } from '../../../helpers/support';

class Tablo extends Component {
    shouldComponentUpdate(nextProps) {
        const { tablo } = this.props;
        return nextProps.tablo !== tablo;
    }

    componentDidUpdate(prevProps) {
        const { tablo, from, to, setNotes, notes } = this.props;
        if (tablo && prevProps.tablo !== tablo) {
            const { lsGet, lsSet } = support;
            const lsNotes = lsGet('notes');
            if (lsNotes) {
                lsNotes.push({ tablo, from, to });
                lsSet('notes', lsNotes);
            } else {
                lsSet('notes', [{ tablo, from, to }]);
            }
            notes.push({ tablo, from, to });
            setNotes(notes);
        }
    }

    render() {
        const { tablo, from, to } = this.props;
        return tablo ? (
            <StTablo>
                <StP>
                    We get
                    <StSpan color="yellow">{tablo}</StSpan>
                    <StSpan>{from.toUpperCase()}</StSpan>
                    from
                    <StSpan>{to.toUpperCase()}</StSpan>
                </StP>
            </StTablo>
        ) : <StTablo />;
    }
}

Tablo.propTypes = {
    tablo: PropTypes.string.isRequired,
    from: PropTypes.string.isRequired,
    to: PropTypes.string.isRequired,
    setNotes: PropTypes.func.isRequired,
    notes: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.string)),
};

export default Tablo;
