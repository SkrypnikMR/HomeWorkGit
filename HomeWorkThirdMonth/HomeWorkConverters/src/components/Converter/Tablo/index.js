import { connect } from 'react-redux';
import { getTablo, getFrom, getTo, getNotes } from '../../../store/converter/selectors';
import { setNotes } from '../../../store/converter/actions';
import Tablo from './Tablo.jsx';

const mapStateToProps = state => ({
    to: getTo(state),
    from: getFrom(state),
    notes: getNotes(state),
    tablo: getTablo(state),
});
const mapDispatchToProps = dispatch => ({ setNotes: payload => dispatch(setNotes(payload)) });

export default connect(mapStateToProps, mapDispatchToProps)(Tablo);
