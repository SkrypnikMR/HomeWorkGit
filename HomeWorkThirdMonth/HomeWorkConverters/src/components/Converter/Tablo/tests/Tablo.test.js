import React from 'react';
import configureStore from 'redux-mock-store';
import { shallowSmart, mountSmart } from '../../../../../__tests__/helpers/testHelper';
import Tablo from '../Tablo';

jest.mock('react-modal');

const mockStore = configureStore();
const store = mockStore({
    isLoading: false,
    converterType: 'lengths',
    from: '',
    to: '',
    quantity: '',
    tablo: '',
    currency: {},
    notes: [],
});

describe('Converter', () => {
    it('Should match snapshot', () => {
        const component = shallowSmart(<Tablo />, store);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render styled__StTablo', () => {
        const component = mountSmart(<Tablo />, store);
        expect(component.find('styled__StTablo')).toHaveLength(1);
    });
    it('should render styled__StTablo with syledP', () => {
        const mockStore = configureStore();
        const store = mockStore({
            isLoading: false,
            converterType: 'lengths',
            from: '',
            to: '',
            quantity: '',
            tablo: 'someTablo',
            currency: {},
            notes: [],
        });
        const component = mountSmart(<Tablo
            tablo='someTablo'
            from='METERS'
            to="VERSTS"
        />, store);
        expect(component.find('styled__StTablo')).toHaveLength(1);
        expect(component.find('styled__StP')).toHaveLength(1);
        expect(component.find('styled__StSpan')).toHaveLength(3);
    });
});
