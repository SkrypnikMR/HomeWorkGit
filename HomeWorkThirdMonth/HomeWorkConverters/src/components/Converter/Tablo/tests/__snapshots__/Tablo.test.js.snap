// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Converter Should match snapshot 1`] = `"<div class=\\"styled__StTablo-sc-1dxfj2m-0 fuzSsk\\"></div>"`;
